index.bs

```diff
@@ -406,73 +413,154 @@ When this method is invoked, the user agent MUST execute the following algorithm
+        else sites will fail on unusual UAs.
+    : {{tokenBinding}}
+    :: The token binding key associated with |callerOrigin| (if any)
+
+        Issue: We need a definition of token binding.
+    : {{ClientData/extensions}}
+    :: <code>{{options}}.{{ScopedCredentialOptions/extensions}}</code>
+
+10. Let |clientDataJSON| be the UTF-8 encoding of the result of calling the
+    original value of {{JSON/stringify|JSON.stringify}} on |clientData|.
+
+    Issue: Some extensions contain ArrayBuffers, which don't stringify well.
+    What's the intent here?
+11. Let |clientDataHash| be the hash of |clientDataJSON| using
+    <code>|clientData|.{{hashAlg}}</code>.
+
+12. Let |issuedRequests| and |currentlyAvailableAuthenticators| be new [=ordered
+    sets=].
+
+13. For each |authenticator| currently available on this platform, if
+    <code>{{options}}.{{ScopedCredentialOptions/attachment}}</code> is not
+    [=present=] or its value matches |authenticator|'s attachment modality,
+    [=set/append=] |authenticator| to |currentlyAvailableAuthenticators|.
+
+14. [=set/For each=] |authenticator| in |currentlyAvailableAuthenticators|:
+    1. Let |excludeList| be a new [=list=].
+    2. [=list/For each=] credential |C| in <code>{{options}}.{{ScopedCredentialOptions/excludeList}}</code>:
+        1. If |C| has an empty {{transports}} list, [=list/append=] it to
+            |excludeList| and [=continue=].
+        2. If |authenticator| is connected over a transport not mentioned in
+            <code>|C|.{{transports}}</code>, the client MAY [=continue=].
+
+            Issue: I'm not sure this captures the intent of the original wording.
+        3. [=list/Append=] |C| to |excludeList|.
+    3. [=In parallel=], invoke the <a>authenticatorMakeCredential</a> operation
+        on |authenticator| with |rpId|, |clientDataHash|,
+        {{accountInformation}}, |normalizedParameters|, |excludeList| and
+        |clientExtensions| as parameters.
+    4. [=set/Append=] |authenticator| to |issuedRequests|.
+
+2. Let |promise| be [=a new promise=]. Return |promise| and start a timer for |adjustedTimeout| milliseconds.
+    Then execute the following steps [=in parallel=]. If any fatal error is encountered in this process other than the
+    ones enumerated below, cancel the timer, [=reject=] |promise| with a DOMException whose name is "{{UnknownError}}", and terminate
+    this algorithm.
 
-6. If the {{ScopedCredentialOptions/extensions}} member of {{options}} is <a>present</a>, process any extensions supported by
-    this client platform, to produce the extension data that needs to be sent to the authenticator. If an error is encountered
-    while processing an extension, skip that extension and do not produce any extension data for it. Call the result of this
-    processing |clientExtensions|.
+    Issue: Should "process" be "algorithm", or does it actually mean an OS
+    process? What kinds of fatal errors are you worried about? I suggest we just
```

Comment:
It seems to me a case can be made that any possible errors are covered by the "If any authenticator returns an error status", but perhaps @vijaybh was worried about various platform-level errors that might percolate up in real life, eg when calling authenticatorCancel (e.g., authnr is now unreachable) and intends for that sentence to be a catch-all.